### PR TITLE
Support lazy setup/teardown

### PR DESCRIPTION
**Pull request by o-lim**
This allows strict or lazy setup/teardown functions to be used explicitly, as well as a command-line flag that defaults all setup/teardown to be lazy, where the default is strict.

### CODE REVIEW

**Comment by o-lim** [comment]
tarruda: Maybe you can see if this PR works for what you want to do with lazy setup/teardown.

File: spec/core_spec.lua [hunk 1]
```diff
@@ -386,3 +386,122 @@ describe('tests unsupported functions in setup/before_each/after_each/teardown',
...
+  describe('another nested describe with tests', function()
+    lazy_setup(function()
+      setup_count = setup_count + 1
+    end)
+
+    it('runs setup for new describe', function()
+      assert.is_equal(3, setup_count)
+    end)
+
+    it('verify teardown ran after preivous describe completes', function()
```
Comment (DorianGray) [inline spec/core_spec.lua]: typo

**Comment by o-lim** [comment]
@tarruda: Sorry, I forgot to put the `@` symbol in an earlier comment.  This is the PR for the lazy setup/teardown.  Give it a try and let me know if this works for you.

**Comment by o-lim** [comment]
Fixed typo and rebased.

**Comment by DorianGray** [comment]
needs another rebase =(

**Comment by o-lim** [comment]
Just did a rebase.

**Comment by tarruda** [comment]
@o-lim thanks, I'm gonna test it. Is there a simple way to install the master branch in my luarocks environment?

**Comment by tarruda** [comment]
Ok I installed using busted-scm-0.rockspec  and it worked perfectly, thanks again!

**Comment by tarruda** [comment]
Do you guys plan to make another release with this feature soon? If so I will hold my PR that adds the `--lazy` command-line option.

**Comment by ajacksified** [comment]
Yeah, we'll cut a release today.

**Comment by tarruda** [comment]
:+1:

**Comment by tarruda** [comment]
Is it possible to install a specific tree from busted repository using luarocks? I can install the master branch like this:

```
luarocks build https://raw.githubusercontent.com/Olivine-Labs/busted/master/busted-scm-0.rockspec
```

but what about installing a specific commit ?

**Comment by mpeterv** [comment]
@tarruda you'll have to clone `busted` repo, checkout to commit you want to install and then run `luarocks make`.

**Comment by tarruda** [comment]
> @tarruda you'll have to clone busted repo, checkout the commit you want to install and then run luarocks make.

:+1: thanks

**Comment by tarruda** [comment]
@ajacksified any chance we can get a new release today?

**Pull request merged**